src/dots.c

```diff
@@ -3,34 +3,40 @@
 #include <R.h>
 #include <Rinternals.h>
 #include <stdio.h>
+#include <stdbool.h>
 
-SEXP ellipsis_dots(SEXP env, SEXP auto_name_) {
-  if (TYPEOF(env) != ENVSXP)
+static SEXP find_dots(SEXP env) {
+  if (TYPEOF(env) != ENVSXP) {
     Rf_errorcall(R_NilValue, "`env` is a not an environment");
-
-  int auto_name = Rf_asLogical(auto_name_);
+  }
 
   SEXP dots = PROTECT(Rf_findVarInFrame3(env, R_DotsSymbol, TRUE));
-  if (dots == R_UnboundValue)
+  if (dots == R_UnboundValue) {
     Rf_errorcall(R_NilValue, "No ... found");
+  }
+
+  UNPROTECT(1);
+  return dots;
+}
+
+SEXP ellipsis_dots(SEXP env, SEXP auto_name_) {
+  int auto_name = Rf_asLogical(auto_name_);
+
+  SEXP dots = PROTECT(find_dots(env));
 
   // Empty dots
   if (dots == R_MissingArg) {
     UNPROTECT(1);
     return Rf_allocVector(VECSXP, 0);
   }
 
-
-  int n = 0;
-  for(SEXP nxt = dots; nxt != R_NilValue; nxt = CDR(nxt)) {
-    n++;
-  }
+  R_len_t n = Rf_length(dots);
 
   SEXP out = PROTECT(Rf_allocVector(VECSXP, n));
   SEXP names = PROTECT(Rf_allocVector(STRSXP, n));
   Rf_setAttrib(out, R_NamesSymbol, names);
 
-  for (int i = 0; i < n; ++i) {
+  for (R_len_t i = 0; i < n; ++i) {
```

Comment:
I really hope we don't need to use this 🤣

Reply:
This is the type for short vector sizes, just for tidiness.